Description: Handle having the compressed data file in /usr/share instead of /etc
Author: Didier Raboud <odyx@debian.org>
Origin: vendor
Last-Update: 2011-01-04
--- a/usb_modeswitch.tcl
+++ b/usb_modeswitch.tcl
@@ -46,6 +46,7 @@
 }
 
 set settings(dbdir)	/etc/usb_modeswitch.d
+set settings(gzdbdir)	/usr/share/usb-modeswitch/
 set bindir	/usr/sbin
 
 set devList1 {}
@@ -537,9 +538,9 @@
 switch $command {
 
 	list {
-		if [file exists $settings(dbdir)/configPack.tar.gz] {
-			Log "Found packed config collection $settings(dbdir)/configPack.tar.gz"
-			if [catch {set configList [exec tar -tzf $settings(dbdir)/configPack.tar.gz 2>/dev/null]} err] {
+		if [file exists $settings(gzdbdir)/configPack.tar.gz] {
+			Log "Found packed config collection $settings(gzdbdir)/configPack.tar.gz"
+			if [catch {set configList [exec tar -tzf $settings(gzdbdir)/configPack.tar.gz 2>/dev/null]} err] {
 				Log "Error: problem opening config package; tar returned\n $err"
 				return {}
 			}
@@ -552,11 +553,11 @@
 		return $configList
 	}
 	config {
-		if [file exists $settings(dbdir)/configPack.tar.gz] {
+		if [file exists $settings(gzdbdir)/configPack.tar.gz] {
 			set settings(tmpConfig) /tmp/usb_modeswitch.current_cfg
-			Log "Extracting config $config from collection $settings(dbdir)/configPack.tar.gz"
+			Log "Extracting config $config from collection $settings(gzdbdir)/configPack.tar.gz"
 			set wc [open $settings(tmpConfig) w]
-			puts -nonewline $wc [exec tar -xzOf $settings(dbdir)/configPack.tar.gz $config 2>/dev/null]
+			puts -nonewline $wc [exec tar -xzOf $settings(gzdbdir)/configPack.tar.gz $config 2>/dev/null]
 			close $wc
 		} else {
 			set settings(tmpConfig) $config
